senders/client_factory.go

```diff
@@ -230,6 +239,19 @@ func TracesPort(port int) Option {
 	}
 }
 
+// Timeout sets the HTTP timeout (in seconds). Defaults to 10 seconds.
+func Timeout(timeout time.Duration) Option {
+	return func(cfg *configuration) {
+		cfg.Timeout = timeout
+	}
+}
+
+func TLSConfigOptions(tlsCfg *tls.Config) Option {
+	return func(cfg *configuration) {
+		cfg.TLSConfigOptions = tlsCfg
```

Comment:
The caller could call TLSConfigOptions() to get an Option, and sometime later change the tlsCfg they pass in. If they do that, it will change the behavior of the Option they created too. Ideally an Option should be immutable.  Consider calling Clone() on tlsCfg to get a defensive copy Like this:  

```
func TLSConfigOptions(tlsCfg *tls.Config) Option {
    tlsCopy := tlsCfg.Clone()
    return func(cfg *configuration) {
        cfg.TLSConfigOptions = tlsCopy
    }
}
```

Reply:
Thank you, @keep94! I made the change, though I'm still trying to understand the concept you described by reading up on Golang (yesterday and today). Thought it might be worth updating you.